google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -476,6 +477,18 @@ static String getServiceAccountProjectId(String credentialsPath) {
     return project;
   }
 
+  /* 
+   * Returns true if the projectId is valid.
+   * projectId must be between 6 and 30 characters
+   * projectId can have lowercase letters, digits or hyphens
+   * and must start with a lowercase letter
+   */
+  private boolean isValidProjectId(String projectId) {
```

Comment:
You asked on IM about test. I think a test is a good idea. You can make this `@InternalApi` so it's visible to test class.